### ISSUE TITLE
Wiki should explain substitution syntax

### ISSUE THREAD

**Issue by MartinHowarth** [google/re2#317]
Hi,

Please can you update the Syntax wiki page to describe the possible substitution patterns? The current Syntax wiki page lists `\1` as not supported with no comment about it being value for substitution. This is really confusing!

For example adding this section to the syntax would be useful (not sure how to make a PR for a wiki page!). I've included the two syntax's I'm aware of, and a note to explain what took me a while to figure out a solution for :) 

-----------------

# Substitution

Some parts of backtracking syntax are supported for use in substitution only.

Syntax  | Action | Notes
------ | ------ | ------
`\1` | Insert capture group 1 |
`\g<1>` | Insert capture group 1 | Useful where the `\1` syntax is ambiguous, for example `\10` should be written `\g<1>0` or `\g<10>` to disambiguate

**Comment by junyer** [comment]
For `RE2::Replace()` et al. in the C++ library, see https://github.com/google/re2/blob/main/re2/re2.h#L444-L453.
**TLDR**: `\0` through `\9` (inclusive) is what's supported.

For the substitution functionality in the Python bindings, see https://docs.python.org/3/library/re.html#re.sub.
**TLDR**: `\1` through `\99` (inclusive) and `\g<...>` are what's supported; `\0` must be written as `\g<0>`.

https://github.com/google/re2/wiki/Syntax is for regular expression syntax, which is a different kettle of lutefisk.
